FraihmworkHealthAndIntegrityApi.yaml

```diff
@@ -856,18 +1276,18 @@ components:
         source:
           type: string
           format: uuid
-          description: The source UUID of the component that is causing the fault. Used when the client can detect failures with other systems or subsystems. When not included, it is assumed that the source is the UUID of this reporting client
+          description: The source UUID of the component that is causing the fault. Used when the client can detect failures with other components or subsystems. When not included, it is assumed that the source is the UUID of this reporting client
         code:
           type: integer
-          description: An integer representation of the fault code. This must be a value that always corresponds to a specific type of fault given a specific version of the system. Implementers are encouraged to give meaningful numbers to faults - for example, faults relating to firmware issues may all fall in the range of 500-599. 
+          description: An integer representation of the fault code. This must be a value that always corresponds to a specific type of fault given a specific version of the component. Implementers are encouraged to give meaningful numbers to faults - for example, faults relating to firmware issues may all fall in the range of 500-599. 
```

Comment:
I like this idea. One step further would be to have code classification ranges suggested to integrators/implementors. This, while it seems like an overstep, would reduce variation in approaches, which is undeniably inevitable.


000-099: Reserved
100-199: System level faults, General
200-299: Software Faults
300-399: Firmware Faults
400-499: Physical Faults
500-599: Hardware Faults
600-699: Sensor Faults
700-799: Data Faults
800-999: [Reserved]

Reply:
This was avoided to allow flexibility for integrating into existing solutions. Although I think recommended values are valuable, I think it is outside the scope of this API spec.